src/service/kafkaRelayPlugin.ts

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: Apache-2.0
+import { Kafka, logLevel, type Producer } from 'kafkajs';
+import { additionalEnvironmentVariables, type Configuration } from '../config';
+import type { LoggerService } from '@tazama-lf/frms-coe-lib';
+import type { Apm } from '@tazama-lf/frms-coe-lib/lib/services/apm';
+import { validateProcessorConfig } from '@tazama-lf/frms-coe-lib/lib/config/processor.config';
+import * as fs from 'fs';
+import type { ITransportPlugin } from '@tazama-lf/frms-coe-lib/lib/interfaces/relay-service/ITransportPlugin';
+
+export default class KafkaRelayPlugin implements ITransportPlugin {
+  private readonly kafka: Kafka;
+  private producer?: Producer;
+  private loggerService?: LoggerService;
+  private apm?: Apm;
+  private readonly configuration: Configuration;
+  private readonly maxInFlight: number;
+
+  constructor() {
+    // Validate and load configuration
+    this.configuration = validateProcessorConfig(additionalEnvironmentVariables) as Configuration;
+
+    const isDev = !this.configuration.nodeEnv || this.configuration.nodeEnv === 'dev';
+
+    const ssl = isDev
+      ? false
+      : {
+          rejectUnauthorized: false,
+          ca: fs.existsSync(this.configuration.KAFKA_TLS_CA!) ? [fs.readFileSync(this.configuration.KAFKA_TLS_CA!)] : [],
+        };
+
+    const parsedMaxInFlight = Number(process.env.maxInFlightRequests);
+
+    if (!parsedMaxInFlight || !Number.isInteger(parsedMaxInFlight) || parsedMaxInFlight <= 0) {
+      throw new Error(`Invalid or missing 'maxInFlightRequests': ${process.env.maxInFlightRequests}`);
+    }
+
+    this.maxInFlight = parsedMaxInFlight;
+
+    this.kafka = new Kafka({
+      clientId: this.configuration.CLIENT_ID ?? 'relay-plugin',
+      brokers: [this.configuration.DESTINATION_TRANSPORT_URL ?? 'localhost:9092'],
+      ssl,
+      logLevel: logLevel.ERROR,
+    });
+  }
```

Comment:
- Specify producer here in constructor (this will remove the need to have producer be optional)
- Update to use suggested KAFKA prefixed configuration variables
```suggestion
    this.kafka = new Kafka({
      clientId: this.configuration.KAFKA_CLIENT_ID ?? 'relay-plugin',
      brokers: [this.configuration.KAFKA_DESTINATION_TRANSPORT_URL ?? 'localhost:9092'],
      ssl,
      logLevel: logLevel.ERROR,
    });

    this.producer = this.kafka.producer({
      maxInFlightRequests: this.configuration.KAFKA_MAX_IN_FLIGHT_REQUESTS,
    });
  }
```